lockfile/src/spdx.rs

```diff
@@ -78,9 +78,9 @@ fn type_from_url(url: &str) -> anyhow::Result<PackageType> {
 }
 
 fn from_purl(pkg_url: &str, pkg_info: &PackageInformation) -> anyhow::Result<Package> {
-    let purl = PackageUrl::from_str(pkg_url)?;
+    let purl = GenericPurl::<String>::from_str(pkg_url)?;
```

Comment:
This seems like an unrelated change. Is there a reason this was done in this PR?

Reply:
Oh, I had made an initial attempt at using PURL and removed it. This got left behind, but I'll remove it.